src/freenet/io/comm/IOStatisticCollector.java

```diff
@@ -4,221 +4,36 @@
 package freenet.io.comm;
 
 import java.net.InetAddress;
-import java.util.Arrays;
-import java.util.Date;
-import java.util.LinkedHashMap;
-import java.util.Map;
 
-import freenet.support.Logger;
-import freenet.support.Logger.LogLevel;
+import freenet.support.transport.ip.IPUtil;
 
 public class IOStatisticCollector {
-	public static final int STATISTICS_ENTRIES = 10;
-	public static final int STATISTICS_DURATION_S = 30;
-	public static final int STATISTICS_DURATION = 1000*STATISTICS_DURATION_S;
-	private long lastrotate;
-	
-	private static boolean logDEBUG;
-	private long totalbytesin;
-	private long totalbytesout;
-	private final LinkedHashMap<String, StatisticEntry> targets;
-	static boolean ENABLE_PER_ADDRESS_TRACKING = false;
-	
-	public IOStatisticCollector() {
-		targets = new LinkedHashMap<String, StatisticEntry>();
-		// TODO: only for testing!!!!
-		// This should only happen once
-		//SNMPAgent.create();
-		//SNMPStarter.initialize();
-		logDEBUG = Logger.shouldLog(LogLevel.DEBUG, this);
-	}
-	
-	public void addInfo(InetAddress addr, int port, int inbytes, int outbytes, boolean isLocal) {
-		try {
-			synchronized (this) {
-				_addInfo(addr, port, inbytes, outbytes, isLocal);
-			}
-		} catch (Throwable t) {
-			t.printStackTrace();
-		}
-	}
-	
-	private void _addInfo(InetAddress addr, int port, int inbytes, int outbytes, boolean isLocal) {
-		rotate();
-		if(ENABLE_PER_ADDRESS_TRACKING) {
-			String key = addr + ":" + port;
-			StatisticEntry entry = targets.get(key);
-			if (entry == null) {
-				entry = new StatisticEntry();
-				targets.put(key, entry);
-			}
-			entry.addData(Math.max(inbytes, 0), Math.max(outbytes, 0));
-		}
-		if(!isLocal) {
-			synchronized(this) {
-				totalbytesout += Math.max(outbytes, 0);
-				totalbytesin += Math.max(inbytes, 0);
-				if(logDEBUG)
-					Logger.debug(IOStatisticCollector.class, "Add("+addr+":"+port+ ',' +inbytes+ ',' +outbytes+" -> "+totalbytesin+" : "+totalbytesout);
-			}
-		}
-	}
-	
-	public void dumpInfo() {
-		synchronized (this) {
-			_dumpInfo();
-		}
-	}
+    private long totalBytesIn;
+    private long totalBytesOut;
 
-	public long[] getTotalIO() {
-		synchronized (this) {
-			return _getTotalIO();
-		}
-	}
-	
-	private long[] _getTotalIO() {
-		long ret[] = new long[2]; 
-		synchronized(this) {
-			ret[0] = totalbytesout;
-			ret[1] = totalbytesin;
-		}
-		return ret;
-	}
-	
-	public int[][] getTotalStatistics() {
-		synchronized (this) {
-			return _getTotalStatistics();
-		}
-	}
+    public void reportReceivedBytes(InetAddress addr, int bytes) {
+        if (isLocal(addr) || bytes <= 0) {
+            return;
```

Comment:
I’m pretty sure that we have calculation errors with overhead when most traffic is local, but that does not have to be solved in this PR.